cmd/api_key/api_key.go

```diff
@@ -0,0 +1,232 @@
+// Licensed to Yugabyte, Inc. under one or more contributor license
+// agreements. See the NOTICE file distributed with this work for
+// additional information regarding copyright ownership. Yugabyte
+// licenses this file to you under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package api_key
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/yugabyte/ybm-cli/cmd/util"
+	ybmAuthClient "github.com/yugabyte/ybm-cli/internal/client"
+	"github.com/yugabyte/ybm-cli/internal/formatter"
+)
+
+var ApiKeyCmd = &cobra.Command{
+	Use:   "api-key",
+	Short: "Manage API Keys",
+	Long:  "Manage API Keys in your YBM account",
+	Run: func(cmd *cobra.Command, args []string) {
+		cmd.Help()
+	},
+}
+
+var listApiKeysCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List API Keys",
+	Long:  `List API Keys in your YBM account`,
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf("could not initiate api client: %s", err.Error())
+		}
+		authApi.GetInfo("", "")
+
+		apiKeyListRequest := authApi.ListApiKeys()
+
+		// if user filters by api key name, add it to the request
+		name, _ := cmd.Flags().GetString("name")
+		if name != "" {
+			apiKeyListRequest = apiKeyListRequest.ApiKeyName(name)
+		}
+
+		// if user filters by key status, add it to the request
+		keyStatus, _ := cmd.Flags().GetString("status")
+		if keyStatus != "" {
+			validStatus := false
+			for _, v := range GetKeyStatusFilters() {
+				if strings.ToUpper(keyStatus) == v {
+					validStatus = true
+					apiKeyListRequest = apiKeyListRequest.Status([]string{v})
+				}
+			}
+			if !validStatus {
+				logrus.Fatalln("Only ACTIVE, EXPIRED, REVOKED status filters are allowed.")
+			}
+		}
+
+		resp, r, err := apiKeyListRequest.Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		apiKeyCtx := formatter.Context{
+			Output: os.Stdout,
+			Format: formatter.NewApiKeyFormat(viper.GetString("output")),
+		}
+
+		if len(resp.GetData()) < 1 {
+			logrus.Info("No API Keys found")
+			return
+		}
+
+		formatter.ApiKeyWrite(apiKeyCtx, resp.GetData())
+	},
+}
+
+func GetKeyStatusFilters() []string {
+	return []string{"ACTIVE", "EXPIRED", "REVOKED"}
```

Comment:
It feels like this should be an enum on the server side? Why is the REST API giving us a string and not that enum?

Reply:
It is an enum and that is used somewhere else but for this request its just taking strings